pkg/kubelet/kubelet_pods.go

```diff
@@ -1785,21 +1795,16 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 			continue
 		}
 
-		if pod.DeletionTimestamp == nil {
-			continue
-		}
-
-		// and if the pod itself is being deleted, then the CRI may have removed the container already and for whatever reason the kubelet missed the exit code
-		// (this seems not awesome).  We know at this point that we will not be restarting the container.
+		klog.InfoS("considering setting termination status", "pod", klog.KObj(pod), "container", container, "oldStatus", oldStatus)
 		status := statuses[container.Name]
 		// if the status we're about to write indicates the default, the Waiting status will force this pod back into Pending.
-		// That isn't true, we know the pod is going away.
+		// That isn't true, we know the pod was restarted.
```

Comment:
This is sort of the crux of the issue -- previously, this logic only toggled on termination, but it's possible for pods to need this logic on restart (e.g. daemonsets, node crash).